sessiontxn/interface.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sessiontxn
+
+import (
+	"fmt"
+
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/stringutil"
+)
+
+// AssertRecordsKey is used to save failPoint invoke records
+// Only for test
```

Comment:
If some variable is only used in the test code, you can write a xxx_test.go file, and then write

```
package sessiontxn

var AssertRecordsKey stringutil.StringerStr = "assertTxnManagerRecords"
```

in that file.